cunumeric/coverage.py

```diff
@@ -136,10 +139,8 @@ def clone_module(
 
     for attr, value in missing.items():
         if isinstance(value, FunctionType):
-            new_globals[attr] = (
-                unimplemented(
-                    value, module_name, reporting=runtime.report_coverage
-                ),
```

Comment:
This comma was a bug

Reply:
Definitely present on current `22.03` I don't see how this isn't causing the existing tests to fail but I have to conclude that we must not be exercising any `unimplemented` function at all?

<details>

```
>>> for k, v in cunumeric.__dict__.items():
...     if isinstance(v, tuple): print(k, v)
... 
show_config (<function show at 0x7f12e00ef3b0>,)
argwhere (<function argwhere at 0x7f11ac02ba70>,)
copyto (<function copyto at 0x7f11ac02b680>,)
lexsort (<function lexsort at 0x7f11ac02bb00>,)
can_cast (<function can_cast at 0x7f11ac02bb90>,)
min_scalar_type (<function min_scalar_type at 0x7f11ac02bc20>,)
result_type (<function result_type at 0x7f11ac02bcb0>,)
isfortran (<function isfortran at 0x7f11ac02bd40>,)
correlate (<function correlate at 0x7f11ac02bdd0>,)
roll (<function roll at 0x7f11ac02be60>,)
rollaxis (<function rollaxis at 0x7f11ac02bef0>,)
moveaxis (<function moveaxis at 0x7f11ac02bf80>,)
cross (<function cross at 0x7f11ac035050>,)
array_equiv (<function array_equiv at 0x7f11ac0350e0>,)
indices (<function indices at 0x7f11ac035170>,)
fromfunction (<function fromfunction at 0x7f11ac035200>,)
isclose (<function isclose at 0x7f11ac035290>,)
isscalar (<function isscalar at 0x7f11ac035320>,)
binary_repr (<function binary_repr at 0x7f11ac0353b0>,)
base_repr (<function base_repr at 0x7f11ac035440>,)
putmask (<function putmask at 0x7f11ac0354d0>,)
flatnonzero (<function flatnonzero at 0x7f11ac035560>,)
shares_memory (<function shares_memory at 0x7f11ac0355f0>,)
may_share_memory (<function may_share_memory at 0x7f11ac035680>,)
alen (<function alen at 0x7f11ac035710>,)
alltrue (<function alltrue at 0x7f11ac0357a0>,)
argpartition (<function argpartition at 0x7f11ac035830>,)
around (<function around at 0x7f11ac0358c0>,)
compress (<function compress at 0x7f11ac035950>,)
cumprod (<function cumprod at 0x7f11ac0359e0>,)
cumproduct (<function cumproduct at 0x7f11ac035a70>,)
cumsum (<function cumsum at 0x7f11ac035b00>,)
ndim (<function ndim at 0x7f11ac035b90>,)
partition (<function partition at 0x7f11ac035c20>,)
product (<function product at 0x7f11ac035cb0>,)
ptp (<function ptp at 0x7f11ac035d40>,)
put (<function put at 0x7f11ac035dd0>,)
resize (<function resize at 0x7f11ac035e60>,)
round_ (<function round_ at 0x7f11ac035ef0>,)
searchsorted (<function searchsorted at 0x7f11ac035f80>,)
size (<function size at 0x7f11ac037050>,)
sometrue (<function sometrue at 0x7f11ac0370e0>,)
std (<function std at 0x7f11ac037170>,)
take (<function take at 0x7f11ac037200>,)
trace (<function trace at 0x7f11ac037290>,)
var (<function var at 0x7f11ac037320>,)
ScalarType (<class 'int'>, <class 'float'>, <class 'complex'>, <class 'int'>, <class 'bool'>, <class 'bytes'>, <class 'str'>, <class 'memoryview'>, <class 'numpy.bool_'>, <class 'numpy.complex64'>, <class 'numpy.complex128'>, <class 'numpy.complex256'>, <class 'numpy.float16'>, <class 'numpy.float32'>, <class 'numpy.float64'>, <class 'numpy.float128'>, <class 'numpy.int8'>, <class 'numpy.int16'>, <class 'numpy.int32'>, <class 'numpy.longlong'>, <class 'numpy.int64'>, <class 'numpy.timedelta64'>, <class 'numpy.datetime64'>, <class 'numpy.object_'>, <class 'numpy.bytes_'>, <class 'numpy.str_'>, <class 'numpy.uint8'>, <class 'numpy.uint16'>, <class 'numpy.uint32'>, <class 'numpy.ulonglong'>, <class 'numpy.uint64'>, <class 'numpy.void'>)
obj2sctype (<function obj2sctype at 0x7f11ac0373b0>,)
sctype2char (<function sctype2char at 0x7f11ac037440>,)
maximum_sctype (<function maximum_sctype at 0x7f11ac0374d0>,)
issctype (<function issctype at 0x7f11ac037560>,)
find_common_type (<function find_common_type at 0x7f11ac0375f0>,)
issubdtype (<function issubdtype at 0x7f11ac037680>,)
datetime_as_string (<function datetime_as_string at 0x7f11ac037710>,)
busday_offset (<function busday_offset at 0x7f11ac0377a0>,)
busday_count (<function busday_count at 0x7f11ac037830>,)
is_busday (<function is_busday at 0x7f11ac0378c0>,)
array2string (<function array2string at 0x7f11ac037950>,)
array_str (<function array_str at 0x7f11ac0379e0>,)
array_repr (<function array_repr at 0x7f11ac037a70>,)
set_string_function (<function set_string_function at 0x7f11ac037b00>,)
set_printoptions (<function set_printoptions at 0x7f11ac037b90>,)
get_printoptions (<function get_printoptions at 0x7f11ac037c20>,)
printoptions (<function printoptions at 0x7f11ac037cb0>,)
format_float_positional (<function format_float_positional at 0x7f11ac037d40>,)
format_float_scientific (<function format_float_scientific at 0x7f11ac037dd0>,)
require (<function require at 0x7f11ac037e60>,)
seterr (<function seterr at 0x7f11ac037ef0>,)
geterr (<function geterr at 0x7f11ac037f80>,)
setbufsize (<function setbufsize at 0x7f11ac042050>,)
getbufsize (<function getbufsize at 0x7f11ac0420e0>,)
seterrcall (<function seterrcall at 0x7f11ac042170>,)
geterrcall (<function geterrcall at 0x7f11ac042200>,)
logspace (<function logspace at 0x7f11ac042290>,)
geomspace (<function geomspace at 0x7f11ac042320>,)
atleast_1d (<function atleast_1d at 0x7f11ac0423b0>,)
atleast_2d (<function atleast_2d at 0x7f11ac042440>,)
atleast_3d (<function atleast_3d at 0x7f11ac0424d0>,)
block (<function block at 0x7f11ac042560>,)
einsum_path (<function einsum_path at 0x7f11ac0425f0>,)
iscomplexobj (<function iscomplexobj at 0x7f11ac042680>,)
isrealobj (<function isrealobj at 0x7f11ac042710>,)
iscomplex (<function iscomplex at 0x7f11ac0427a0>,)
isreal (<function isreal at 0x7f11ac042830>,)
nan_to_num (<function nan_to_num at 0x7f11ac0428c0>,)
real_if_close (<function real_if_close at 0x7f11ac042950>,)
typename (<function typename at 0x7f11ac0429e0>,)
asfarray (<function asfarray at 0x7f11ac042a70>,)
mintypecode (<function mintypecode at 0x7f11ac042b00>,)
asscalar (<function asscalar at 0x7f11ac042b90>,)
common_type (<function common_type at 0x7f11ac042c20>,)
ravel_multi_index (<function ravel_multi_index at 0x7f11ac042cb0>,)
unravel_index (<function unravel_index at 0x7f11ac042d40>,)
ix_ (<function ix_ at 0x7f11ac042dd0>,)
fill_diagonal (<function fill_diagonal at 0x7f11ac042e60>,)
diag_indices (<function diag_indices at 0x7f11ac042ef0>,)
diag_indices_from (<function diag_indices_from at 0x7f11ac042f80>,)
select (<function select at 0x7f11ac048050>,)
piecewise (<function piecewise at 0x7f11ac0480e0>,)
trim_zeros (<function trim_zeros at 0x7f11ac048170>,)
iterable (<function iterable at 0x7f11ac048200>,)
percentile (<function percentile at 0x7f11ac048290>,)
diff (<function diff at 0x7f11ac048320>,)
gradient (<function gradient at 0x7f11ac0483b0>,)
angle (<function angle at 0x7f11ac048440>,)
unwrap (<function unwrap at 0x7f11ac0484d0>,)
disp (<function disp at 0x7f11ac048560>,)
rot90 (<function rot90 at 0x7f11ac0485f0>,)
extract (<function extract at 0x7f11ac048680>,)
place (<function place at 0x7f11ac048710>,)
asarray_chkfinite (<function asarray_chkfinite at 0x7f11ac0487a0>,)
average (<function average at 0x7f11ac048830>,)
digitize (<function digitize at 0x7f11ac0488c0>,)
cov (<function cov at 0x7f11ac048950>,)
corrcoef (<function corrcoef at 0x7f11ac0489e0>,)
median (<function median at 0x7f11ac048a70>,)
sinc (<function sinc at 0x7f11ac048b00>,)
hamming (<function hamming at 0x7f11ac048b90>,)
hanning (<function hanning at 0x7f11ac048c20>,)
bartlett (<function bartlett at 0x7f11ac048cb0>,)
blackman (<function blackman at 0x7f11ac048d40>,)
kaiser (<function kaiser at 0x7f11ac048dd0>,)
trapz (<function trapz at 0x7f11ac048e60>,)
i0 (<function i0 at 0x7f11ac048ef0>,)
add_newdoc (<function add_newdoc at 0x7f11ac048f80>,)
meshgrid (<function meshgrid at 0x7f11ac04c050>,)
delete (<function delete at 0x7f11ac04c0e0>,)
insert (<function insert at 0x7f11ac04c170>,)
interp (<function interp at 0x7f11ac04c200>,)
quantile (<function quantile at 0x7f11ac04c290>,)
apply_over_axes (<function apply_over_axes at 0x7f11ac04c320>,)
expand_dims (<function expand_dims at 0x7f11ac04c3b0>,)
apply_along_axis (<function apply_along_axis at 0x7f11ac04c440>,)
kron (<function kron at 0x7f11ac04c4d0>,)
get_array_wrap (<function get_array_wrap at 0x7f11ac04c560>,)
take_along_axis (<function take_along_axis at 0x7f11ac04c5f0>,)
put_along_axis (<function put_along_axis at 0x7f11ac04c680>,)
broadcast_to (<function broadcast_to at 0x7f11ac04c710>,)
broadcast_arrays (<function broadcast_arrays at 0x7f11ac04c7a0>,)
broadcast_shapes (<function broadcast_shapes at 0x7f11ac04c830>,)
diagflat (<function diagflat at 0x7f11ac04c8c0>,)
fliplr (<function fliplr at 0x7f11ac04c950>,)
flipud (<function flipud at 0x7f11ac04c9e0>,)
tri (<function tri at 0x7f11ac04ca70>,)
vander (<function vander at 0x7f11ac04cb00>,)
histogram2d (<function histogram2d at 0x7f11ac04cb90>,)
mask_indices (<function mask_indices at 0x7f11ac04cc20>,)
tril_indices (<function tril_indices at 0x7f11ac04ccb0>,)
tril_indices_from (<function tril_indices_from at 0x7f11ac04cd40>,)
triu_indices (<function triu_indices at 0x7f11ac04cdd0>,)
triu_indices_from (<function triu_indices_from at 0x7f11ac04ce60>,)
fix (<function fix at 0x7f11ac04cef0>,)
isneginf (<function isneginf at 0x7f11ac04cf80>,)
isposinf (<function isposinf at 0x7f11ac054050>,)
pad (<function pad at 0x7f11ac0540e0>,)
poly (<function poly at 0x7f11ac054170>,)
roots (<function roots at 0x7f11ac054200>,)
polyint (<function polyint at 0x7f11ac054290>,)
polyder (<function polyder at 0x7f11ac054320>,)
polyadd (<function polyadd at 0x7f11ac0543b0>,)
polysub (<function polysub at 0x7f11ac054440>,)
polymul (<function polymul at 0x7f11ac0544d0>,)
polydiv (<function polydiv at 0x7f11ac054560>,)
polyval (<function polyval at 0x7f11ac0545f0>,)
polyfit (<function polyfit at 0x7f11ac054680>,)
issubclass_ (<function issubclass_ at 0x7f11ac054710>,)
issubsctype (<function issubsctype at 0x7f11ac0547a0>,)
deprecate (<function deprecate at 0x7f11ac054830>,)
deprecate_with_doc (<function deprecate_with_doc at 0x7f11ac0548c0>,)
get_include (<function get_include at 0x7f11ac054950>,)
info (<function info at 0x7f11ac0549e0>,)
source (<function source at 0x7f11ac054a70>,)
who (<function who at 0x7f11ac054b00>,)
lookfor (<function lookfor at 0x7f11ac054b90>,)
byte_bounds (<function byte_bounds at 0x7f11ac054c20>,)
safe_eval (<function safe_eval at 0x7f11ac054cb0>,)
ediff1d (<function ediff1d at 0x7f11ac054d40>,)
intersect1d (<function intersect1d at 0x7f11ac054dd0>,)
setxor1d (<function setxor1d at 0x7f11ac054e60>,)
union1d (<function union1d at 0x7f11ac054ef0>,)
setdiff1d (<function setdiff1d at 0x7f11ac054f80>,)
in1d (<function in1d at 0x7f11ac05b050>,)
isin (<function isin at 0x7f11ac05b0e0>,)
savetxt (<function savetxt at 0x7f11ac05b170>,)
loadtxt (<function loadtxt at 0x7f11ac05b200>,)
genfromtxt (<function genfromtxt at 0x7f11ac05b290>,)
ndfromtxt (<function ndfromtxt at 0x7f11ac05b320>,)
mafromtxt (<function mafromtxt at 0x7f11ac05b3b0>,)
recfromtxt (<function recfromtxt at 0x7f11ac05b440>,)
recfromcsv (<function recfromcsv at 0x7f11ac05b4d0>,)
load (<function load at 0x7f11ac05b560>,)
loads (<function loads at 0x7f11ac05b5f0>,)
save (<function save at 0x7f11ac05b680>,)
savez (<function savez at 0x7f11ac05b710>,)
savez_compressed (<function savez_compressed at 0x7f11ac05b7a0>,)
packbits (<function packbits at 0x7f11ac05b830>,)
unpackbits (<function unpackbits at 0x7f11ac05b8c0>,)
fromregex (<function fromregex at 0x7f11ac05b950>,)
nansum (<function nansum at 0x7f11ac05b9e0>,)
nanmax (<function nanmax at 0x7f11ac05ba70>,)
nanmin (<function nanmin at 0x7f11ac05bb00>,)
nanargmax (<function nanargmax at 0x7f11ac05bb90>,)
nanargmin (<function nanargmin at 0x7f11ac05bc20>,)
nanmean (<function nanmean at 0x7f11ac05bcb0>,)
nanmedian (<function nanmedian at 0x7f11ac05bd40>,)
nanpercentile (<function nanpercentile at 0x7f11ac05bdd0>,)
nanvar (<function nanvar at 0x7f11ac05be60>,)
nanstd (<function nanstd at 0x7f11ac05bef0>,)
nanprod (<function nanprod at 0x7f11ac05bf80>,)
nancumsum (<function nancumsum at 0x7f11ac062050>,)
nancumprod (<function nancumprod at 0x7f11ac0620e0>,)
nanquantile (<function nanquantile at 0x7f11ac062170>,)
histogram (<function histogram at 0x7f11ac062200>,)
histogramdd (<function histogramdd at 0x7f11ac062290>,)
histogram_bin_edges (<function histogram_bin_edges at 0x7f11ac062320>,)
bmat (<function bmat at 0x7f11ac0623b0>,)
mat (<function asmatrix at 0x7f11ac062440>,)
asmatrix (<function asmatrix at 0x7f11ac0624d0>,)
round (<function round_ at 0x7f11ac062560>,)
```
</details>

Reply:
Thins working correctly on this branch:
```
>>> for k, v in cunumeric.__dict__.items():
...     if isinstance(v, tuple): print(k, v)
... 
ScalarType (<class 'int'>, <class 'float'>, <class 'complex'>, <class 'int'>, <class 'bool'>, <class 'bytes'>, <class 'str'>, <class 'memoryview'>, <class 'numpy.bool_'>, <class 'numpy.complex64'>, <class 'numpy.complex128'>, <class 'numpy.complex256'>, <class 'numpy.float16'>, <class 'numpy.float32'>, <class 'numpy.float64'>, <class 'numpy.float128'>, <class 'numpy.int8'>, <class 'numpy.int16'>, <class 'numpy.int32'>, <class 'numpy.longlong'>, <class 'numpy.int64'>, <class 'numpy.timedelta64'>, <class 'numpy.datetime64'>, <class 'numpy.object_'>, <class 'numpy.bytes_'>, <class 'numpy.str_'>, <class 'numpy.uint8'>, <class 'numpy.uint16'>, <class 'numpy.uint32'>, <class 'numpy.ulonglong'>, <class 'numpy.uint64'>, <class 'numpy.void'>)
kernel_version (5, 10)
```